src/engine/controls/cuecontrol.cpp

```diff
@@ -136,6 +136,27 @@ CueControl::~CueControl() {
     qDeleteAll(m_hotcueControls);
 }
 
+void CueControl::process(const double,
+        mixxx::audio::FramePos currentPosition,
+        const int) {
+    for (const auto& pCue : std::as_const(m_hotcueControls)) {
+        if (pCue->getStatus() != HotcueControl::Status::Active ||
+                pCue->getCue()->getType() != mixxx::CueType::Jump) {
+            continue;
+        }
+        if (pCue->getPosition() > m_lastProcessedPosition &&
+                pCue->getPosition() <= currentPosition) {
+            auto delta = pCue->getPosition() - currentPosition;
```

Comment:
That does not work well, because the jump is quantized by the engine buffer size. Normally you want to jump beat exact and which can be somewhere within the buffer. 

So we need something like the looping engine. 

How about actually using the looping engine. Isn't a jump just the same as a loop which disables itself after one iteration.

Reply:
Why does it not work? This is already a simplified version of `LoopingControl::process`, except that we do not need `adjustedPositionInsideAdjustedLoop`, but can seek to the "target" directly, (corrected with the engine delay to prevent desync, as done with that `delta` var)

There is a few drawback I can think about using the looping engine:
- Loop engine expects start < end, but jump may have from > to. Allowing start > end in `LoopingControl` sounds wrong
- They maybe be a loop defined (but not active) and using the `LoopingControl` will override it for a "one frame" loop (or will have to deal with state restoration, which sounds overkill)

> Isn't a jump just the same as a loop which disables itself after one iteration.

To some degree, except it is more simple that a loop, and doesn't need to complexity around the loop (no concept of seekMode for example, or direction). Disabling it at the next iteration implies that is had to be enable at some point, which is why making the most of `EngineControl` seems a better approach. The engine is already able to perform jumps, so I'm not sure what is the value of using the LoopingControl for a simple unique unidirectional jump

Reply:
The seeking code in LoopingControl::process handled only the exceptional cases when adjusting the loop size. The looping itself is done here: 
https://github.com/mixxxdj/mixxx/blob/e32b1ede66169982a421ecdb2135773790e991e4/src/engine/readaheadmanager.cpp#L49

Your current implementation does not work for reverse playing and jumps before the track, and small jumps, < delta. 

It took quite some iterations to have the looping code rock solid and I have not a big interest to repeat these iterations with beat jumps again. So let's consider what is the best way for trying that code. 

> Loop engine expects start < end, but jump may have from > to. Allowing start > end in LoopingControl sounds wrong

I think we have discussed earlier to allow forward loops. The engine don't mind if it is a repeated jump backward or a single jump forward. 
However we need to take care not to create surprises when we allow to edit jumps with looping controls. So I agree to not reuse them in the first attempt. 

> They maybe be a loop defined (but not active) and using the LoopingControl will override it for a "one frame" loop (or will have to deal with state restoration, which sounds overkill)

We have this issue already with saved loops. 
For example I would love to have a saved loop at the end of the track to extend a short outro that is always active. Maybe we can solve both in one run.

Reply:
> Your current implementation does not work for reverse playing and jumps before the track, and small jumps, < delta.

It does need to, because of the unidirectional aspect of a jump. IMO, you wouldn't expect to jump back to the `from` position if you play reverse above `to`

> For example I would love to have a saved loop at the end of the track to extend a short outro that is always active

Definitely would like to see this behaviour too as I could use it as well. But perhaps that is out of scope for this feature?

Reply:
For the reverse case, my expectation was that it jumps form the "to" position to "from" just as if you would play backwards a forward recording of the jump. But It is also OK for the first version to not jump at all in reverse, which is is probably not yet the case . (not tested)

Reply:
The idea of this feature is to "automate" a `beatjump_forward` so not sure it would make sense to "undo" this jump if playing reverse. 
I guess we could make it work with a minimal amount of change, without using the LoopingControl if that the behaviour with would expect.